framework/src/problems/FEProblemBase.C

```diff
@@ -2202,35 +2202,22 @@ FEProblemBase::getSampler(const std::string & name, THREAD_ID tid)
 
 bool
 FEProblemBase::duplicateVariableCheck(const std::string & var_name,
-                                      const FEType & type,
                                       bool is_aux)
 {
   SystemBase * curr_sys_ptr = _nl.get();
   SystemBase * other_sys_ptr = _aux.get();
-  std::string error_prefix = "";
   if (is_aux)
   {
     curr_sys_ptr = _aux.get();
     other_sys_ptr = _nl.get();
-    error_prefix = "Aux";
   }
 
   if (other_sys_ptr->hasVariable(var_name))
     mooseError("Cannot have an auxiliary variable and a nonlinear variable with the same name: ",
                var_name);
 
   if (curr_sys_ptr->hasVariable(var_name))
-  {
-    const Variable & var =
-        curr_sys_ptr->system().variable(curr_sys_ptr->system().variable_number(var_name));
-    if (var.type() != type)
```

Comment:
This removed check is happening in libMesh's `add_variable` anyways, so I don't see the point of repeating it here (except wanting to give the user a MOOSE flavored error message).